addons/html_editor/static/src/utils/sanitize.js

```diff
@@ -34,17 +35,23 @@ export function initElementForEdition(element, options = {}) {
  * Properly close common XML-like self-closing elements to avoid HTML parsing
  * issues.
  *
- * @param {string} content
- * @returns {string}
+ * @param {string | ReturnType<markup>} content
+ * @returns {ReturnType<markup>}
  */
 export function fixInvalidHTML(content) {
     if (!content) {
         return content;
     }
     // TODO: improve the regex to support nodes with data-attributes containing
     // `/` and `>` characters.
-    const regex = /<\s*(a|strong|t|span)[^<]*?\/\s*>/g;
-    return content.replace(regex, (match, g0) => match.replace(/\/\s*>/, `></${g0}>`));
+    const regex = /<\s*(a|strong|t|span)([^<]*?)\/\s*>/g;
+    return htmlReplace(content, regex, (match, tag, attributes) => {
+        // markup: content is either already markup or escaped in htmlReplace
+        attributes = markup(attributes);
+        // markup: tag can only take one of the pre-defined values in the regex
+        tag = markup(tag);
```

Comment:
reading this again, I don't think markup is necessary here: tag is just a static string with just letters, there is no risk of double-escaping anything if markup is not present